### ISSUE TITLE
Class-level thresholds calculated by validation set?

### ISSUE THREAD

**Issue by windygoo** [Zhimin-C/Confid-SSL-Det#1]
It is probably better to maintain a queue on the unlablled set, the validation set is sometimes unavailable, especially in the case of little labeled data.

**Comment by Zhimin-C** [comment]
Hi, 

As there is no testing set in those detection datasets, we just leverage the validation set without any annotations to calculate the threshold. In the classification task, our results indicate that using unlabeled data from the training set achieves comparable results as the unlabeled validation set. I think this is because the training data and the validation data have the same distribution. You can also try to utilize the unlabeled part from the training data to calculate the threshold in the detection task.

Zhimin Chen

**Comment by windygoo** [comment]
Great work! 
Another question. Adopting the average confidence scores as thresholds is clearly larger than the actual thresholds. The mapping function x / (2 - x) is designed to pull down the estimated thresholds? If I try your method on other datasets, should I adapt the formulation of this mapping function?

**Comment by Zhimin-C** [comment]
Yes, the mapping function is designed to pull down the estimated thresholds. As we have tired this mapping function on 4 3D datasets in both detection and classification tasks, I think this function is general enough for 3D tasks. However, if you want to implement it in the 2D dataset, you may need to adapt the mapping function.
